test/integration/tests/pauseState.ts

```diff
@@ -20,50 +19,36 @@ describe('e2e-pause-state', function () {
 
   before(async function () {
     // Setup test environment and get contracts
-    const version = 1;
     deployAddress = (await ethers.getSigners())[0].address;
     if (!deployAddress) throw new Error(`No deploy address!`);
 
-    doLogging = Boolean(process.env.LOGGING);
-
     const config = {
-      logging: doLogging,
+      logging: Boolean(process.env.LOGGING),
       deployAddress: deployAddress,
-      version: version
+      version: 1
     };
 
     e2eCoord = new TestEndtoEndCoordinator(config, proposals);
 
-    doLogging && console.log(`Loading environment...`);
+    config.logging && console.log(`Loading environment...`);
     ({ contracts } = await e2eCoord.loadEnvironment());
-    doLogging && console.log(`Environment loaded.`);
+    config.logging && console.log(`Environment loaded.`);
   });
 
-  it('should reflect on-chain protocol pause state', async function () {
-    const pausedContractNames = Object.keys(pauseStateConfig);
-
-    for (let i = 0; i < pausedContractNames.length; i += 1) {
-      const pausedContractName = pausedContractNames[i];
-      const expectedContractPauseConfig = pauseStateConfig[pausedContractName];
-
-      // Get the on-chain contract
-      const onChainContract = contracts[pausedContractName];
+  it('Procotol State Configuration', async function () {
+    for (const contractName of StateConfigEntryNames) {
+      console.log(`Checking ${contractName} state configs...`);
 
-      const onChainPauseState = await onChainContract.paused();
-      expect(onChainPauseState).to.be.equal(expectedContractPauseConfig.paused, 'paused() state incorrect');
+      const stateConfigEntry = StateConfig[contractName];
+      const contract = contracts[contractName];
 
-      // Validate PSM related pausing
-      if (expectedContractPauseConfig.redeemPaused) {
-        expect(await onChainContract.redeemPaused()).to.be.equal(
-          expectedContractPauseConfig.redeemPaused,
-          'redeemPaused() state incorrect'
-        );
-      }
+      for (const keyName of Object.keys(stateConfigEntry)) {
+        const expectedValue = stateConfigEntry[keyName as keyof typeof stateConfigEntry];
+        const actualValue = await contract[keyName]();
```

Comment:
would this work for non boolean types?

Reply:
It should, but I'll add in some test cases against string and a number/big number types just to make sure it does.

Reply:
would be cool to allow addresses as expected values, for instance `'{pcvGuardian}'` instead of the `'0x.....'` value

Reply:
> would be cool to allow addresses as expected values, for instance `'{pcvGuardian}'` instead of the `'0x.....'` value

You can actually do this by just using a var for any value in any of the key-value pairs in the state config file.